common/thrift_router.cpp

```diff
@@ -51,6 +51,7 @@ bool parseHost(const std::string& str, common::detail::Host* host,
     return false;
   }
   auto group = (tokens.size() == 3 ? tokens[2] : "");
+  host->az = group;
```

Comment:
This is actually a combination of availability zone + placement group. So I don't think az is the right name for this. Did you want to filter by AZ or PG or both?

Reply:
Actually it's ok to only be AZ aware and not look into PG. Do you suggest fishing out the AZ from the string?